### PR TITLE
fix(java): rename test to be picked up by native profile

### PR DESCRIPTION
**Pull request by mpeddada1**
A couple of changes to note:
1) The maven-surefire-plugin defined in [datastore-v1-proto-client/pom.xml](https://github.com/googleapis/java-datastore/pull/723/files#diff-18fb0d587c1b12f0cb5729d02ed0d791ec5b5ea14844da40e21dbf9cbc681908)  was overwriting the one defined in the native profile brought in from java-shared-config, therefore, resulting in the following error: ```Test configuration file wasn't found. Make sure that test execution wasn't skipped.```.  Tested this change locally to verify that integration tests are excluded when unit tests are run. Also, java-shared-config brings in the same config defined in the POM: https://github.com/googleapis/java-shared-config/blob/b5e05c6726c60a7ad7c0d9588b4fab69a0b4e6b7/pom.xml#L87

2) Replaced usages of `@Rule public ExpectedException thrown = ExpectedException.none();` with Truth. (background: go/cloud-java-native-image#truth-vs-hamcrest-use-truth)

3) Renamed test that uses EasyMock `ITDatastoreTest` -> `DatastoreTest` as mocking is incompatible with native image compilation atm.

### CODE REVIEW

**Comment by mpeddada1** [comment]
The Kokoro Native Image job is now passing

**Comment by mpeddada1** [comment]
However, the unit tests seem to be failing with the following error:
```
Trying to download...
Downloading Java 17.0.3+7 (Zulu) from https://cdn.azul.com/zulu/bin/zulu17.34.19_1-ca-jdk17.0.3-linux_x64.tar.gz ...
Error: Unexpected HTTP response: 404
```

File: datastore-v1-proto-client/src/main/resources/META-INF/native-image/native-image.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+Args = --enable-url-protocols=https,http
```
Comment (suztomo) [inline datastore-v1-proto-client/src/main/resources/META-INF/native-image/native-image.properties:1]: Does this library use HTTP (not HTTPS) somewhere in users' environment? (What's the error without this argument?)

If the request is only for testing, then this file should be under test.

**Comment by suztomo** [comment]
@kolea2 This is a followup from our conversation after the Wednesday meeting.

File: datastore-v1-proto-client/src/main/resources/META-INF/native-image/native-image.properties (hunk 1 above)
Comment (mpeddada1) [reply datastore-v1-proto-client/src/main/resources/META-INF/native-image/native-image.properties:1]: Looking at the stacktrace, this is coming from `com.google.datastore.v1.client.RemoteRpc`: 
```
 Caused by: java.lang.IllegalArgumentException: java.net.MalformedURLException: Accessing an URL protocol that was not enabled. The URL protocol https is supported but not enabled by default. It must be enabled by adding the --enable-url-protocols=https option to the native-image command.
       com.google.api.client.http.GenericUrl.parseURL(GenericUrl.java:679)
       com.google.api.client.http.GenericUrl.<init>(GenericUrl.java:125)
       com.google.api.client.http.GenericUrl.<init>(GenericUrl.java:108)
       com.google.datastore.v1.client.RemoteRpc.resolveURL(RemoteRpc.java:161)
       com.google.datastore.v1.client.RemoteRpc.<init>(RemoteRpc.java:62)
```
Normally this configuration would come from gax, but this module doesn't import it so we had to add this setting in.

**Review by suztomo — APPROVED** [review]
If the allow HTTP is passed in GAX already, then no need to reject it here.

File: datastore-v1-proto-client/src/main/resources/META-INF/native-image/native-image.properties (hunk 1 above)
Comment (mpeddada1) [reply datastore-v1-proto-client/src/main/resources/META-INF/native-image/native-image.properties:1]: The error is coming from `RemoteRpc.resolveURL` but it's used in [`RemoteRpc#call` ]( https://github.com/googleapis/java-datastore/blob/957a6b5b940edc476344a38cc6038a4a4851039f/datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/RemoteRpc.java#L86) which is later referenced in `Datastore`. For this reason, I think we would still need it in the main configuration?

**Review by kolea2 — APPROVED** [review]

**Comment by mpeddada1** [comment]
Thank you all so much!

**Pull request merged**